File: 1-TwoSum/1-TwoSum.py
# Last updated: 07/04/2026, 11:43:58
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        note = {}

        for i in range(len(nums)) :

            d = target - nums[i] 
            if d in note :
                return [note[d], i ]

            note[nums[i]] = i     
    

File: 1-TwoSum/1-TwoSum.java
// Last updated: 09/04/2026, 12:52:53
1class Solution {
2    public int[] twoSum(int[] nums, int target) {
3        HashMap <Integer,Integer> mp = new HashMap<>();
4        for ( int i  =0;i<nums.length;i++){
5            int j = target-nums[i];
6            if ( mp.containsKey(j)){
7                return new int[] { mp.get(j),i};
8            }
9            mp.put(nums[i], i);
10            }
11            return new int[] {-1,-1};
12    }
13}